keanu-python/keanu/vertex/iff.py

```diff
@@ -0,0 +1,39 @@
+from typing import Union
+
+from keanu.infer_type import get_type_of_value
+from .base import Boolean
+from .vertex_casting import do_inferred_vertex_cast
+from .base import Vertex, vertex_constructor_param_types
+from .generated import BooleanIf, DoubleIf, IntegerIf, ConstantBoolean, ConstantInteger, ConstantDouble
+
+
+def __cast_to_vertex(input: vertex_constructor_param_types) -> Vertex:
+    return do_inferred_vertex_cast({bool: ConstantBoolean, int: ConstantInteger, float: ConstantDouble}, input)
+
+
+from keanu.vartypes import tensor_arg_types
+
+
+def If(predicate: Union[tensor_arg_types, Vertex], thn: Union[tensor_arg_types, Vertex],
+       els: Union[tensor_arg_types, Vertex]) -> Vertex:
+    predicate = __cast_to_vertex(predicate)
+    thn = __cast_to_vertex(thn)
+    els = __cast_to_vertex(els)
```

Comment:
Do you need this?  Won't just calling the constructor for the If Vertex make this happen anyway?